src/Datadog.Trace/HttpOverStreams/HttpContent/IHttpContent.cs

```diff
@@ -0,0 +1,14 @@
+using System.IO;
+using System.Threading.Tasks;
+
+namespace HttpOverStream
+{
+    internal interface IHttpContent
+    {
+        long? Length { get; }
+
+        void CopyTo(Stream destination);
```

Comment:
I started out doing everything `async`, but it was getting in the way of debugging, so I switched back to all synchronous until I got things working. We should probably move everything back to `async` and remove the synchronous versions.